lib/v2/5eplay/index.js

```diff
@@ -0,0 +1,83 @@
+const cheerio = require('cheerio');
+const zlib = require('zlib');
+const got = require('@/utils/got');
+const { parseDate } = require('@/utils/parse-date');
+const { getAcwScV2ByArg1 } = require('./utils');
+
+module.exports = async (ctx) => {
+    const rootUrl = 'https://csgo.5eplay.com/';
+    const apiUrl = `${rootUrl}api/article?page=1&type_id=0&time=0&order_by=0`;
+    const articleUrl = `${rootUrl}article`;
+
+    const { data: response } = await got({
+        method: 'get',
+        url: apiUrl,
+    });
+
+    const expiredTime = 25 * 60 * 1000;
+
+    let acw_sc__v2 = '';
+    // get arg1
+    await ctx.cache.tryGet(
+        articleUrl,
+        async () => {
+            // Zlib Z_BUF_ERROR, should close decompress
+            const detailResponse = await got(articleUrl, {
+                decompress: false,
+            });
+
+            const obj_unzip = zlib.createUnzip({
+                chunkSize: 20 * 1024,
+            });
+            obj_unzip.write(detailResponse.body);
+            const arg1 = await new Promise((resolve) => {
+                obj_unzip.on('data', (data) => {
+                    resolve(data.toString().match(/var arg1='(.*?)';/)[1]);
+                });
+            });
+            acw_sc__v2 = getAcwScV2ByArg1(arg1);
```

Comment:
I think it's possible to do it with loop
```js
            for await (const data of obj_unzip) {
                const strData = data.toString();
                const matches = strData.match(/var arg1='(.*?)';/);
                if (matches) {
                    acw_sc__v2 = getAcwScV2ByArg1(matches[1]);
                    break;
                }
            }
```

Reply:
Awesome, I didn't know it was possible to write code like this and it works